### PR TITLE
Remake cluster size autoscaling scale to zero test

### PR DESCRIPTION
**Pull request by aleksandra-malinowska**
This PR affects only cluster size autoscaling test suite. Changes:
* check whether autoscaling for is enabled by looking for a node group with a given max number of nodes instead of min as the field is omitted if value is 0
* split scale to zero test into GKE & GCE versions, add GKE-specific setup and verification

### CODE REVIEW

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 1]
```diff
@@ -921,6 +940,17 @@ func getPoolNodes(f *framework.Framework, poolName string) []*v1.Node {
 	return nodes
 }
 
+func getPoolSize(f *framework.Framework, poolName string) int {
+	size := 0
+	nodeList := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+	for _, node := range nodeList.Items {
+		if poolLabel := node.Labels["cloud.google.com/gke-nodepool"]; poolLabel == poolName {
```
Comment (MaciekPytel) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: cloud.google.com/gke-nodepool -> const?

Comment (aleksandra-malinowska) [reply test/e2e/autoscaling/cluster_size_autoscaling.go]: done

**Pull request merged**